[BUG] Building fails in timezone UTC+3

## Bug summary

Building from Arch Linux AUR I get this error:

```
testTimeZone (__main__.TimeTest.testTimeZone)
Test stringtotime on two strings straddling timezones ... FAIL

======================================================================
FAIL: testTimeZone (__main__.TimeTest.testTimeZone)
Test stringtotime on two strings straddling timezones
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/james/.cache/yay/rdiff-backup/src/rdiff-backup-2.2.5/testing/timetest.py", line 153, in testTimeZone
    self.assertEqual(f(invf(f(s1))), f(s1))
AssertionError: 1112514303 != 1112510703

----------------------------------------------------------------------
Ran 12 tests in 2.105s

FAILED (failures=1)
==> ERROR: A failure occurred in check().
    Aborting...
 -> error making: rdiff-backup-exit status 4
 -> Failed to install the following packages. Manual intervention is required:
rdiff-backup - exit status 4
```

It seems that there is 3600 (one hour?) difference (incorrect timezone data somewhere?). If I set "timedatectl set-timezone UTC" then that error doesn't happen.

## Version, Python, Operating System

```yaml
exec:
  api_version:
    actual: 0
    default: 200
    max: 201
    min: 200
  argv:
  - /usr/bin/rdiff-backup
  - info
  parsed: null
  version: 2.2.5
python:
  executable: /usr/bin/python
  name: cpython
  version: 3.11.3
system:
  fs_encoding: utf-8
  platform: Linux-6.1.38-1-lts-x86_64-with-glibc2.37
```

## rdiff-backup call

I'm trying to build rdiff-backup as a regular user (from wheel group), like it's supposed:

```
yay -S rdiff-backup
```

## What happened and what did you expect?

Build fails for 'Europe/Moscow' (UTC+3).

## More information

It seems that there is 3600 (one hour?) difference (incorrect timezone data somewhere?).

If I set "timedatectl set-timezone UTC" then that error doesn't happen.

--- a/src/rdiff_backup/Time.py
+++ b/src/rdiff_backup/Time.py
@@ -314,28 +314,15 @@ def _get_tzd(timeinseconds=None):
     """
     if timeinseconds is None:
         timeinseconds = time.time()
-    dst_in_effect = time.daylight and time.localtime(timeinseconds)[8]
-    if dst_in_effect:
-        offset = -time.altzone / 60
-    else:
-        offset = -time.timezone / 60
-    if offset > 0:
-        prefix = "+"
-    elif offset < 0:
-        prefix = "-"
-    else:
-        return "Z"  # time is already in UTC
-
+    tzd = time.strftime("%z", time.localtime(timeinseconds))
     if Globals.use_compatible_timestamps:
         time_separator = '-'
     else:
         time_separator = ':'
-    hours, minutes = list(map(abs, divmod(offset, 60)))
-    assert 0 <= hours <= 23, (
-        "Hours {hrs} must be between 0 and 23".format(hrs=hours))
-    assert 0 <= minutes <= 59, (
-        "Minutes {mins} must be between 0 and 59".format(mins=minutes))
-    return "%s%02d%s%02d" % (prefix, hours, time_separator, minutes)
+    if tzd == "+0000":
+        return "Z"
+    else:
+        return tzd[:3] + time_separator + tzd[3:]
 
 
 def _tzd_to_seconds(tzd):

